src/action.ts

```diff
@@ -57,6 +57,12 @@ export default async function main() {
     ? appendToPreReleaseTag
     : currentBranch;
 
+  // Sanitize indentifier if it contains a semver
+  if (!!valid(identifier)) {
+    let ver = String(valid(identifier));
+    identifier.replace(ver, ver.replace('.', '_'));
+  }
```

Comment:
You can't use `_` in semver. Also, you shouldn't expect `identifier` to contain a valid semver, it's rarely a case since most often it's the branch name. 

A better approach would be to sanitize `identifier` unconditionally, according to [semver spec](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions)

```suggestion
// Sanitize identifier according to https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions
let sanitized = identifier.replace(/[^a-zA-Z0-9-]/g, '-');
```